service/src/test/java/org/sunbird/metrics/actors/CourseMetricsActorTest.java

```diff
@@ -125,6 +129,76 @@ public void testCourseProgressMetricsWithInvalidPeriod() {
     Assert.assertEquals("INVALID_PERIOD", e.getCode());
   }
 
+  @Test
+  public void testCourseProgressMetricsV2WithInvalidBatchId() {
+
+    TestKit probe = new TestKit(system);
```

Comment:
There is a template except - mocking rules and expected output - please reduce the code-duplicacy - create common test-scenario - mock in test-method then call testscenario with expected output...